### PR TITLE
Feature/thumbnails

### PR DESCRIPTION
**Pull request by felix-schwarz**
Add thumbnail support:

bildschirmfoto 2018-04-28 um 23 01 02

BUGS & IMPROVEMENTS

- [X] (1) Browsing again to a folder does not display thumbnail https://github.com/owncloud/ios-app/pull/32#issuecomment-389415369 [FIXED]
- [X] (2) Scroll down https://github.com/owncloud/ios-app/pull/32#issuecomment-389422892 [FIXED]

### CODE REVIEW

File: ownCloud/Client/ClientItemCell.swift [hunk 1]
```diff
@@ -90,6 +101,34 @@ class ClientItemCell: ThemeTableViewCell {
 			self.accessoryType = .none
 		}
 
+		if item.thumbnailAvailability != .none {
+			let displayThumbnail = { (thumbnail: OCItemThumbnail?) in
+				thumbnail?.requestImage(for: thumbnailSize, scale: 0, withCompletionHandler: { (thumbnail, error, _, image) in
+					if error == nil {
+						DispatchQueue.main.async {
+							if image != nil {
+								if self.item?.versionIdentifier == thumbnail?.versionIdentifier {
```
Comment (pablocarmu) [inline ownCloud/Client/ClientItemCell.swift]: Also in Swift you can chain multiple if statements with a `,`. Don't know which option is more clear to you if the one with three if statements or chain them like: 
```Swift
if error == nil, image != nil, self.item?.versionIdentifier == thumbnail?.versionIdentifier {
  OnMainThread {
       self.iconView.image = image
   }
}
```

File: ownCloud/Client/ClientItemCell.swift [hunk 2]
```diff
@@ -90,6 +101,34 @@ class ClientItemCell: ThemeTableViewCell {
...
+									self.iconView.image = image
+								}
+							}
+						}
+					}
+				})
+			}
+
+			if let thumbnail = item.thumbnail {
+				_ = displayThumbnail(thumbnail)
```
Comment (pablocarmu) [inline ownCloud/Client/ClientItemCell.swift]: I think the displayThumbnail closure is taking the unused `thumbnail?.requestImage(...)` Bool return as it's own return parameter and this is why it's saying to you `Result of call is unused, but produces 'Bool?'` when you later  call  `displayThumbnail(thumbnail)` if you don't assign it's return value to some variable ( in this case to a `_`)

You can get rid of this `_` assignment if you code it here:
```Swift
_ = thumbnail?.requestImage(for: thumbnailSize, scale: 0, withCompletionHandler: { (thumbnail, error, _, image)
```
then you can call the `displayThumbnail(thumbnail)` without any assignment: 
```Swift
displayThumbnail(thumbnail)
```

File: ownCloud/Client/ClientItemCell.swift [hunk 3]
```diff
@@ -90,6 +101,34 @@ class ClientItemCell: ThemeTableViewCell {
 			self.accessoryType = .none
 		}
 
+		if item.thumbnailAvailability != .none {
+			let displayThumbnail = { (thumbnail: OCItemThumbnail?) in
+				thumbnail?.requestImage(for: thumbnailSize, scale: 0, withCompletionHandler: { (thumbnail, error, _, image) in
+					if error == nil {
+						DispatchQueue.main.async {
```
[the remark this answers is not in the document]
Comment (felix-schwarz) [reply ownCloud/Client/ClientItemCell.swift]: Good point! Thanks.

File: ownCloud/Client/ClientItemCell.swift (hunk 1 above)
Comment (felix-schwarz) [reply ownCloud/Client/ClientItemCell.swift]: Good idea. I made that change.

File: ownCloud/Client/ClientItemCell.swift (hunk 2 above)
Comment (felix-schwarz) [reply ownCloud/Client/ClientItemCell.swift]: Thanks. Made that change as well.

**Comment by jesmrec** [comment]
### (1)

1. Browse into a folder with pics (thumbnails downloaded and displayed)
2. Browse back to parent folder
3. Browse again to folder in 1.

Current: Thumbnails not displayed
Expected: Thumbnails displayed

**Comment by jesmrec** [comment]
### (2)

1. Open the app with pics in root folder (thumbnails displayed)
2. Scroll down
3. Scroll up

Current: Thumbnails in 1. not displayed
Expected: Thumbnails displayed

gif2

**Comment by jesmrec** [comment]
Formats that thumbnails are not displayed: TIFF, SVG.

**Comment by felix-schwarz** [comment]
The latest commit fixes the issues (1) and (2).

Regarding TIFF and SVG: right now, the app should request thumbnails for all items.

Chances are the server you're testing against doesn't provide thumbnails for this format.

**Comment by jesmrec** [comment]
All stuff fixed. Approved.

**Pull request merged**